src/main/java/org/folio/qm/service/impl/CacheServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package org.folio.qm.service.impl;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.stereotype.Service;
+import org.springframework.web.context.request.async.DeferredResult;
+
+import org.folio.qm.service.CacheService;
+
+@Log4j2
+@Service
+@SuppressWarnings("rawtypes")
+public class CacheServiceImpl implements CacheService<DeferredResult> {
+
+  private final Cache<String, DeferredResult> cache = CacheBuilder.newBuilder().maximumSize(1000).build();
```

Comment:
I'd make any cache settings configurable, in particular max size

Have you considered Spring Cache as abstraction for caching?